yetanotherco/aligned_layer
# fix(audit): Periodically Remove  oldTask data from aggregator

# This PR:
Removes Task data from aggregator response maps after a response has been responded to or expired.

closes #977 

#To Test:
Setup local dev net as per the guide

```
make anvil_start_with_block_time
```

```
 make aggregator_start
```

```
make batcher_start_local
```

(optional:)
```
make operator_register_and_start
```

And leave a task sender running until the end of the test
```
make batcher_send_burst_groth16
```

 Everything should work as normally.

Also, every 120 seconds, you should see the log: `Cleaning finalized tasks from maps` , with some extra information about which tasks are being removed from the Aggregator. These are the tasks being removed, all older than 10 blocks of age.

In prod this variables change to bigger numbers, but the flow is the same.

Note: in dev, first iteration may be small because of the constraints, but the second iteration should include more tasks.
